app/src/main/kotlin/com/flxrs/dankchat/preferences/notifications/highlights/HighlightsScreen.kt

```diff
@@ -461,6 +493,134 @@ private fun UserHighlightItem(
+                        label = { Text(stringResource(R.string.badge)) },
+                        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
+                        maxLines = 1,
+                    )
+                } else {
+                    var name = ""
+                    when (item.badgeName) {
+                        "broadcaster"-> name = stringResource(R.string.badge_broadcaster)
+                        "admin"-> name = stringResource(R.string.badge_admin)
+                        "staff"-> name = stringResource(R.string.badge_staff)
+                        "moderator"-> name = stringResource(R.string.badge_moderator)
+                        "partner"-> name = stringResource(R.string.badge_verified)
+                        "vip"-> name = stringResource(R.string.badge_vip)
+                        "founder"-> name = stringResource(R.string.badge_founder)
+                        "subscriber"-> name = stringResource(R.string.badge_subscriber)
+                    }
+                    OutlinedTextField(
+                        modifier = Modifier.fillMaxWidth(),
+                        value = name,
+                        onValueChange = {},
+                        readOnly = true,
+                        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
+                        maxLines = 1,
+                    )
+                }
+                FlowRow(
+                    modifier = Modifier.padding(top = 8.dp),
+                    verticalArrangement = Arrangement.Center,
+                ) {
+                    CheckboxWithText(
+                        text = stringResource(R.string.enabled),
+                        checked = item.enabled,
+                        onCheckedChange = { onChanged(item.copy(enabled = it)) },
+                        modifier = modifier.padding(end = 8.dp),
+                    )
+                    CheckboxWithText(
+                        text = stringResource(R.string.create_notification),
+                        checked = item.createNotification,
+                        onCheckedChange = { onChanged(item.copy(createNotification = it)) },
+                        enabled = item.enabled && item.notificationsEnabled,
+                    )
+                }
+                val defaultColor = ContextCompat.getColor(LocalContext.current, R.color.color_mention_highlight)
+                val color = item.customColor ?: defaultColor
+                var showColorPicker by remember { mutableStateOf(false) }
+                var selectedColor by remember(color) { mutableIntStateOf(color) }
+                OutlinedButton(
+                    onClick = { showColorPicker = true },
+                    enabled = item.enabled,
+                    colors = ButtonDefaults.outlinedButtonColors(contentColor = Color(color)),
+                    content = { Text(String.format("#%08x", color)) },
```

Comment:
For the color picker part I would wait for your other PR #1007 to be merged and potentially introduce the adaptions then here as well.